templates/decays.cpp.in

```diff
@@ -0,0 +1,294 @@
+   BOOST_ASSERT_MSG(x<=1, "This function");
+   if (is_zero(x-1)) {
+      return 0.;
+   } else if (1-x < 1e-6) {
+      return 39*(-1 + x) + (75*Sqr(-1 + x))/2. - 6*Power3(-1 + x);
+   } else {
+    return 3.*(1. - 8.*x + 20.*x*x)/std::sqrt(4.*x-1.) * std::acos((3.*x-1.)/(2.*std::pow(x, 3./2.)))
+    - (1.-x)/(2.*x) * (2. - 13.*x + 47.*x*x)
+    - 3./2. * (1. - 6.*x + 4.*x*x) * std::log(x);
+   }
+}
+
+} // anonymous namespace
+
+@partialWidthSpecializationDefs@
+
+template <>
+double CLASSNAME::get_partial_width<H, A, A>(
+      const context_base& context,
+      const typename field_indices<H>::type& in_idx,
+      const typename field_indices<A>::type& out1_idx,
+      const typename field_indices<A>::type& out2_idx) const
+{
+   const auto amp = calculate_amplitude<H, A, A>(context, in_idx, out1_idx, out2_idx);
+   const double mH = context.physical_mass<H>(in_idx);
+   constexpr double ps {1./(8.*Pi)};
+   constexpr double ps_symmetry {1./2.};
+   const double color_gen_sqr = squared_color_generator<H, A, A>();
+   const double flux = 0.5/mH;
+   const auto res = flux * color_gen_sqr * ps * ps_symmetry * amp.square();
+
+   return res;
+}
+
+template <>
+double CLASSNAME::get_partial_width<H, A, Z>(
+      const context_base& context,
+      const typename field_indices<H>::type& in_idx,
+      const typename field_indices<A>::type& out1_idx,
+      const typename field_indices<Z>::type& out2_idx) const
+{
+   const double mH = context.physical_mass<H>(in_idx);
+   const double mA = context.physical_mass<A>(out1_idx);
+   const double mZ = context.physical_mass<Z>(out2_idx);
+
+   const double ps {1./(8.*Pi) * sqrt(KallenLambda(1., Sqr(mA/mH), Sqr(mZ/mH)))};
+   constexpr double ps_symmetry {1.};
+   const double color_gen_sqr = squared_color_generator<H, A, Z>();
+   const double flux = 0.5/mH;
+
+   const auto amp = calculate_amplitude<H, A, Z>(context, in_idx, out1_idx, out2_idx);
+
+   const auto res = flux * color_gen_sqr * ps * ps_symmetry * amp.square();
+
+   return res;
+}
+
+template <>
+double CLASSNAME::get_partial_width<H, G, G>(
+      const context_base& context,
+      const typename field_indices<H>::type& in_idx,
+      const typename field_indices<G>::type& out1_idx,
+      const typename field_indices<G>::type& out2_idx) const
+{
+   const auto amp = calculate_amplitude<H, G, G>(context, in_idx, out1_idx, out2_idx);
+   const double mH = context.physical_mass<H>(in_idx);
+   constexpr double ps {1./(8.*Pi)};
+   constexpr double ps_symmetry {1./2.};
+   const double color_fact = squared_color_generator<H, G, G>();
+   const double flux = 0.5/mH;
+
+   double result = flux * color_fact * ps * ps_symmetry * amp.square();
+
+   // higher order corrections to CP-even Higgs
+   if (!info::is_CP_violating_Higgs_sector) {
+      // higher order QCD corrections
+
+      // number of active light flavours
+      unsigned int Nf;
+      const double mtpole {qedqcd.displayPoleMt()};
+      if (mH > 5 && mH < mtpole) {
```

Comment:
This check: `mH > 5` looks very strange. (There is also a similar check for `mA` in another function.) Can we write it in a more direct (and const) form like so:
~~~.cpp
const unsigned Nf = mH < mtpole ? 5 : 6;
~~~
?

Reply:
For some default param cards in FS the higgs mass is < 5 GeV.

Reply:
I see. Although, this is a strange result. :)

TBH, I find this logic confusing, because we're throwing an exception if `mH <= 5` and if the Higgs sector is not CP violating. But if the Higgs sector is CP violating and `mH <= 5` we're not throwing?

It seems to me we should
* either *always* throw an exception (independent of the CP properties of the Higgs sector) or
* don't include the QCD corrections if `mH <= 5`

I'd propose to do something like:
~~~.cpp
// minimum required Higgs mass for inclusion of QCD corrections
const double min_Higgs_mass = 5.0;

// higher order corrections to CP-even Higgs
if (mH > min_Higgs_mass && !info::is_CP_violating_Higgs_sector) {
   // higher order QCD corrections
   const unsigned Nf = mH < mtpole ? 5 : 6;
   // ...
}
~~~